### PR TITLE
feat(windows terminal): supported

### PR DESCRIPTION
**Pull request by thuanpham2311**
#67 
image
- i not sure how to get the theme name?

### CODE REVIEW

**Comment by thuanpham2311** [comment]
- i running windows 11 with lua 5.4.2
- i don't know what happen
image

**Comment by thuanpham2311** [comment]
> * i running windows 11 with lua 5.4.2
> * i don't know what happen
>   image

image

**Comment by thuanpham2311** [comment]
> * i running windows 11 with lua 5.4.2
> * i don't know what happen
>   image

image

**Comment by mcchrish** [comment]
Probably because `extras/windowsTerminal` does not exist. You have to create the directory.

Also, let's go with `windows_terminal` instead.

**Comment by thuanpham2311** [comment]
@mcchrish, Already for review

File: lua/zenbones/shipwright/runners/windows_terminal.lua [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+"brightGreen": "$bright_green",
+"brightPurple": "$bright_magenta",
+"brightRed": "$bright_red",
+"brightWhite": "$bright_white",
+"brightYellow": "$bright_yellow",
+"cursorColor": "$cursor_bg",
+"cyan": "$cyan",
+"foreground": "$fg",
+"green": "$green",
+"name": "zenbonse",
```
Comment (mcchrish) [inline lua/zenbones/shipwright/runners/windows_terminal.lua:17]: Name should be the colorscheme name.

File: lua/zenbones/shipwright/runners/windows_terminal.lua [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+"purple": "$magenta",
+"red": "$red",
+"selectionBackground": "$selection_bg",
+"white": "$white",
+"yellow": "$yellow"
+}
+]]
```
Comment (mcchrish) [inline lua/zenbones/shipwright/runners/windows_terminal.lua:24]: I think we should order them by context, not alphabetically.

File: lua/zenbones/shipwright/runners/windows_terminal.lua [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+---@diagnostic disable: undefined-global
+-- selene: allow(undefined_variable)
+run(
+	{ specs, p, term },
+	transform.colorscheme_to_term_colors,
+	function(colors)
+		local text = helpers.apply_template(template, colors)
+		return { text }
+	end,
+	{ prepend, "// This file is auto-generated by shipwright.nvim (don't copy this line)" },
```
Comment (mcchrish) [inline lua/zenbones/shipwright/runners/windows_terminal.lua:37]: Since it's json, we probably should not include this line.

**Review by mcchrish — CHANGES_REQUESTED** [review]

File: lua/zenbones/shipwright/runners/windows_terminal.lua (hunk 1 above)
Comment (thuanpham2311) [reply lua/zenbones/shipwright/runners/windows_terminal.lua]: we have any variable for this $???

Comment (mcchrish) [reply lua/zenbones/shipwright/runners/windows_terminal.lua]: Yes, it just `name`. Checkout the kitty runner where it uses the name. It should pretty much the same.

Comment (thuanpham2311) [reply lua/zenbones/shipwright/runners/windows_terminal.lua]: - i read the shipwright.nvim [code](https://github.com/rktjmp/shipwright.nvim/blob/a3fee9725346eba3eacba36989c906a550a32ae7/lua/shipwright/transform/contrib/kitty.lua?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L168)
```lua
  local replaced = helpers.split_newlines(helpers.apply_template(template, colors))
  local kept = {}
  for _, line in ipairs(replaced) do
    if not string.match(line, "%$") then
      table.insert(kept, line)
    end
  end

  return kept
 ```
 - i think this code will convert $name, i try ~40m write it in windows terminal runner but not worked 😅

Comment (mcchrish) [reply lua/zenbones/shipwright/runners/windows_terminal.lua]: I meant the kitty runner in zenbones. https://github.com/mcchrish/zenbones.nvim/blob/99da53ada1715f1f5758b936f03ce5dab0dfe5cc/lua/zenbones/shipwright/runners/kitty.lua#L5-L19

File: lua/zenbones/shipwright/runners/windows_terminal.lua [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+}
+]]
+
+local helpers = require "shipwright.transform.helpers"
+
+---@diagnostic disable: undefined-global
+-- selene: allow(undefined_variable)
+run(
+	{ specs, p, term },
+	transform.colorscheme_to_term_colors,
```
Comment (mcchrish) [inline lua/zenbones/shipwright/runners/windows_terminal.lua:32]: Like this:

```suggestion
	transform.colorscheme_to_term_colors,
	function(colors)
		return vim.tbl_extend("keep", colors, { name = name })
	end
```

See: https://github.com/mcchrish/zenbones.nvim/blob/main/lua/zenbones/shipwright/runners/kitty.lua#L5-L19

**Review by mcchrish — CHANGES_REQUESTED** [review]

File: lua/zenbones/shipwright/runners/windows_terminal.lua (hunk 4 above)
Comment (thuanpham2311) [reply lua/zenbones/shipwright/runners/windows_terminal.lua:32]: 😅

**Comment by thuanpham2311** [comment]
@mcchrish i think it worked

**Review by mcchrish — APPROVED** [review]

**Comment by mcchrish** [comment]
Thanks!

**Comment by mcchrish** [comment]
@thuanpham2311 I think it would be great if the transformer is ported up to shipwright so other colorschemes can make use of it as well. https://github.com/rktjmp/shipwright.nvim/tree/master/lua/shipwright/transform/contrib
What  do you think?

**Comment by thuanpham2311** [comment]
> @thuanpham2311 I think it would be great if the transformer is ported up to shipwright so other colorschemes can make use of it as well. https://github.com/rktjmp/shipwright.nvim/tree/master/lua/shipwright/transform/contrib What do you think?

- It is great, i will create PR to shipwright

**Pull request merged**